File: src/main/java/cs/android/view/list/CSListRowView.java
package cs.android.view.list;

import cs.android.json.CSJSONData;
import cs.android.viewbase.CSLayoutId;
import cs.java.callback.CSRunWith;

/**
 * Created by renetik on 05/01/18.
 */

public class CSListRowView<DataType extends CSJSONData> extends CSRowView<CSListRow<DataType>> {
    private final CSRunWith<CSListRowView<DataType>> _onLoad;

    public CSListRowView(CSListController parent, CSLayoutId layout, CSRunWith<CSListRowView<DataType>> onLoad) {
        super(parent, layout);
        _onLoad = onLoad;
    }

    protected void onLoad(CSListRow<DataType> row) {
        super.onLoad(row);
        _onLoad.run(this);
    }

}

File: library/renetik-android/src/main/java/renetik/android/view/list/CSListRowView.kt
package renetik.android.view.list

import renetik.android.json.CSJsonData
import renetik.android.view.base.CSLayoutId

class CSListRowView<DataType : CSJsonData>(
        parent: CSListController<CSListRow<DataType>, *>, layout: CSLayoutId,
        private val onLoadData: (CSListRowView<DataType>).(CSListRow<DataType>) -> Unit)
    : CSRowView<CSListRow<DataType>>(parent, layout) {

    override fun onLoad(row: CSListRow<DataType>) = onLoadData.invoke(this, row)
}
